<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CryptoTransformsWindow</class>
 <widget class="QWidget" name="CryptoTransformsWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>579</width>
    <height>593</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Cryptographic Transformations</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../grypto_ui.qrc">
    <normaloff>:/grypto/icons/main.png</normaloff>:/grypto/icons/main.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="gb_mode">
     <property name="title">
      <string>Mode</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <item>
       <widget class="QComboBox" name="cb_mode">
        <item>
         <property name="text">
          <string>Encrypt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Decrypt</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Hash</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Pass-Through</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="sw_mode">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <widget class="QWidget" name="page_passwordDetails">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Key Info:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="lbl_keyStatus">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_changePassword">
            <property name="toolTip">
             <string>Change the password used to encrypt or decrypt.</string>
            </property>
            <property name="text">
             <string>Change</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btn_testPassword">
            <property name="toolTip">
             <string>Guess the password that is currently set.</string>
            </property>
            <property name="text">
             <string>Test</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_hashSelection">
         <layout class="QFormLayout" name="formLayout_3">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_6">
            <property name="text">
             <string>Hash Type:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cb_hashType"/>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="gb_source">
     <property name="title">
      <string>Source Data</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QComboBox" name="cb_source">
        <property name="toolTip">
         <string>Source device</string>
        </property>
        <item>
         <property name="text">
          <string>File</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>String</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="sw_source">
        <widget class="QWidget" name="page">
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="1">
           <widget class="QLineEdit" name="le_src"/>
          </item>
          <item row="0" column="2">
           <widget class="QToolButton" name="toolButton_2">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Select source file:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="3">
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_2">
         <layout class="QFormLayout" name="formLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="1" column="0" colspan="2">
           <widget class="QPlainTextEdit" name="te_src">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>1</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Input text</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Select Encoding:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cb_sourceEncoding">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The encoding determines how the input text will be interpreted.&lt;/p&gt;&lt;p&gt;UTF-8/ASCII - Regular text like what you see here.&lt;/p&gt;&lt;p&gt;Hex - Each nibble (4 bytes) must be represented by the ASCII characters 0-9 and a-f (case insensitive).&lt;/p&gt;&lt;p&gt;Base 64 - The input is in base 64 notation.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="currentIndex">
             <number>2</number>
            </property>
            <item>
             <property name="text">
              <string>Hex</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Base 64</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>UTF-8 / ASCII</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="gb_dest">
     <property name="title">
      <string>Destination</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QComboBox" name="cb_dest">
        <property name="toolTip">
         <string>Destination device</string>
        </property>
        <item>
         <property name="text">
          <string>File</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>String</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="sw_dest">
        <widget class="QWidget" name="page_3">
         <layout class="QGridLayout" name="gridLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Select destination file:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="le_dest"/>
          </item>
          <item row="0" column="2">
           <widget class="QToolButton" name="toolButton">
            <property name="text">
             <string>...</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="3">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page_4">
         <layout class="QFormLayout" name="formLayout">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="1" column="0" colspan="2">
           <widget class="QTextBrowser" name="te_dest">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>1</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Output data</string>
            </property>
            <property name="textInteractionFlags">
             <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cb_destEncoding">
            <property name="toolTip">
             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;The encoding determines how the transformed data will be shown.&lt;/p&gt;&lt;p&gt;Hex - Each nibble (4 bytes) will be printed with the ASCII characters 0-9 and a-f.&lt;/p&gt;&lt;p&gt;Base 64 - Converts the binary data to base 64 representation.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <item>
             <property name="text">
              <string>Hex</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Base 64</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>UTF-8 / ASCII</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Select Encoding:</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="pushButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="MinimumExpanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Start operation</string>
     </property>
     <property name="text">
      <string>Do it</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../grypto_ui.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>cb_source</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>sw_source</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>90</x>
     <y>169</y>
    </hint>
    <hint type="destinationlabel">
     <x>81</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cb_dest</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>sw_dest</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>384</y>
    </hint>
    <hint type="destinationlabel">
     <x>180</x>
     <y>404</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>CryptoTransformsWindow</receiver>
   <slot>_do_it()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>280</x>
     <y>576</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>270</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton_2</sender>
   <signal>clicked()</signal>
   <receiver>CryptoTransformsWindow</receiver>
   <slot>_select_file_src()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>532</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>447</x>
     <y>133</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>toolButton</sender>
   <signal>clicked()</signal>
   <receiver>CryptoTransformsWindow</receiver>
   <slot>_select_file_dest()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>539</x>
     <y>441</y>
    </hint>
    <hint type="destinationlabel">
     <x>447</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_changePassword</sender>
   <signal>clicked()</signal>
   <receiver>CryptoTransformsWindow</receiver>
   <slot>_change_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>281</x>
     <y>101</y>
    </hint>
    <hint type="destinationlabel">
     <x>253</x>
     <y>366</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_testPassword</sender>
   <signal>clicked()</signal>
   <receiver>CryptoTransformsWindow</receiver>
   <slot>_test_password()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>424</x>
     <y>96</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>363</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cb_mode</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>CryptoTransformsWindow</receiver>
   <slot>_mode_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>141</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel">
     <x>177</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_do_it()</slot>
  <slot>_select_file_src()</slot>
  <slot>_select_file_dest()</slot>
  <slot>_change_password()</slot>
  <slot>_test_password()</slot>
  <slot>_mode_changed(int)</slot>
 </slots>
</ui>
